WooCommerce/WooCommerceTests/ViewRelated/Orders/Order Creation/NewOrderViewModelTests.swift

```diff
@@ -102,7 +102,55 @@ class NewOrderViewModelTests: XCTestCase {
         viewModel.createOrder()
 
         // Then
-        XCTAssertEqual(viewModel.notice, NewOrderViewModel.NoticeFactory.createOrderCreationErrorNotice())
+        XCTAssertEqual(viewModel.notice, NewOrderViewModel.NoticeFactory.createOrderErrorNotice())
+    }
+
+    func test_view_model_fires_error_notice_when_order_sync_fails() {
+        // Given
+        let stores = MockStoresManager(sessionManager: .testingInstance)
+        let synchronizer = RemoteOrderSynchronizer(siteID: sampleSiteID, stores: stores)
+        let viewModel = NewOrderViewModel(siteID: sampleSiteID, stores: stores, enableRemoteSync: true)
+        let expectation = expectation(description: "Order sync failed")
+        stores.whenReceivingAction(ofType: OrderAction.self) { action in
+            switch action {
+            case let .createOrder(_, _, onCompletion):
+                onCompletion(.failure(NSError(domain: "Error", code: 0)))
+                expectation.fulfill()
+            default:
+                XCTFail("Received unsupported action: \(action)")
+            }
+        }
```

Comment:
nit: We have the `waitForExpectations { }` helper!

Reply:
Thanks! I missed this helper before.